[SPARK-10169] [SQL] [BRANCH-1.3] Partial aggregation's plan is wrong when a grouping expression is used as an argument of the aggregate fucntion

yhuai opened the pull request:
https://issues.apache.org/jira/browse/SPARK-10169

yhuai committed:
Use transformDown.

Conflicts:
	sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #41419 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41419/consoleFull) for   PR 8380 at commit [`f7f6945`](https://github.com/apache/spark/commit/f7f69458989967f0f18f58eaf6ffe051f8eaefdc).
 * This patch merges cleanly.

SparkQA commented:
[Test build #41419 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41419/consoleFull) for   PR 8380 at commit [`f7f6945`](https://github.com/apache/spark/commit/f7f69458989967f0f18f58eaf6ffe051f8eaefdc).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41419/
Test FAILed.

cloud-fan committed:
don't trim the top level Alias

Conflicts:
	sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

SparkQA commented:
[Test build #1681 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1681/consoleFull) for   PR 8380 at commit [`f7f6945`](https://github.com/apache/spark/commit/f7f69458989967f0f18f58eaf6ffe051f8eaefdc).
 * This patch merges cleanly.

SparkQA commented:
[Test build #1681 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1681/consoleFull) for   PR 8380 at commit [`f7f6945`](https://github.com/apache/spark/commit/f7f69458989967f0f18f58eaf6ffe051f8eaefdc).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #41431 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41431/consoleFull) for   PR 8380 at commit [`059c956`](https://github.com/apache/spark/commit/059c9566a8efd0c65f2dcbbd114a7397c656d22f).
 * This patch merges cleanly.

SparkQA commented:
[Test build #41431 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41431/consoleFull) for   PR 8380 at commit [`059c956`](https://github.com/apache/spark/commit/059c9566a8efd0c65f2dcbbd114a7397c656d22f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41431/
Test FAILed.

SparkQA commented:
[Test build #1682 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1682/consoleFull) for   PR 8380 at commit [`059c956`](https://github.com/apache/spark/commit/059c9566a8efd0c65f2dcbbd114a7397c656d22f).
 * This patch merges cleanly.

SparkQA commented:
[Test build #1682 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1682/consoleFull) for   PR 8380 at commit [`059c956`](https://github.com/apache/spark/commit/059c9566a8efd0c65f2dcbbd114a7397c656d22f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

marmbrus commented:
LGTM

marmbrus commented:
Thanks guys, merged!
